api/host_volume_claims.go

```diff
@@ -0,0 +1,70 @@
+// Copyright (c) HashiCorp, Inc.
+// SPDX-License-Identifier: MPL-2.0
+
+package api
+
+import "net/url"
+
+// TaskGroupHostVolumeClaim associates a task group with a host volume ID. It's
+// used for stateful deployments, i.e., volume requests with "sticky" set to
+// true.
+type TaskGroupHostVolumeClaim struct {
+	ID            string `mapstructure:"id" hcl:"id"`
+	Namespace     string `mapstructure:"namespace" hcl:"namespace"`
+	JobID         string `mapstructure:"job_id" hcl:"job_id"`
+	TaskGroupName string `mapstructure:"task_group_name" hcl:"task_group_name"`
+	AllocID       string `mapstructure:"alloc_id" hcl:"alloc_id"`
+	VolumeID      string `mapstructure:"volume_id" hcl:"volume_id"`
+	VolumeName    string `mapstructure:"volume_name" hcl:"volume_name"`
```

Comment:
This API doesn't support any kind of HCL inputs (or writes of any kind using this object), so I think we can skip these tags:

```suggestion
	ID            string `mapstructure:"id"`
	Namespace     string `mapstructure:"namespace"`
	JobID         string `mapstructure:"job_id"`
	TaskGroupName string `mapstructure:"task_group_name"`
	AllocID       string `mapstructure:"alloc_id"`
	VolumeID      string `mapstructure:"volume_id"`
	VolumeName    string `mapstructure:"volume_name"`
```